Sources/SWBCore/SpecImplementations/ProductTypes.swift

```diff
@@ -649,7 +649,7 @@ public final class XCTestBundleProductTypeSpec : BundleProductTypeSpec, @uncheck
         var (tableOpt, warnings, errors) = super.overridingBuildSettings(scope, platform: platform)
         var table = tableOpt ?? MacroValueAssignmentTable(namespace: scope.namespace)
 
-        let isDeviceBuild = platform?.isDeploymentPlatform == true && platform?.identifier != "com.apple.platform.macosx"
+        let isDeviceBuild = platform?.isDeploymentPlatform == true && platform?.name != scope.evaluate(BuiltinMacros.HOST_PLATFORM)
```

Comment:
As an aside, this seems fundamentally flawed. There's no guarantee you're running the resulting binaries on the same host that built them, so it seems the host platform should enable this too.

Why simulators are excluded I'm not sure.